### ISSUE TITLE
Warn about all CAPS acronym/initialisms in names

### ISSUE THREAD

**Issue by yogurtearl** [detekt/detekt#3200]
## Expected Behavior of the rule

Various `*Naming` rules should warn about violations of this advice: 

https://kotlinlang.org/docs/reference/coding-conventions.html#choosing-good-names
```
When using an acronym as part of a declaration name, capitalize it if it consists of two letters (IOStream); capitalize only the first letter if it is longer (XmlFormatter, HttpInputStream).
```

## Context

Don't want to have to manually watch for this. :)

**Comment by BraisGabin** [comment]
So, a list of acronyms and check that those string of chars never appear together and all CAPS. That list should be configurable. We can list some acronyms to be added as default values in that list:

- http
- url
- uri

**Comment by yogurtearl** [comment]
More the 4 capital letters in a row is generally violating this rule, so could be as simple as just warning on 4 capital letters in a row? and then maybe having an configuable allowlist of exceptions to the rule? 

Normally `ABCThingy` would trigger a warning, but if I put `"ABC"` on my allowlist, then it wouldn't be a warning.

**Comment by schalkms** [comment]
@yogurtearl 
What did you find missing in the standard configuration that allows regular expressions achieving exactly this?

@BraisGabin 
I'm honestly not a fan of introducing new configuration blocks for this edge cases, when it could be solved with existing functionality.
As I see it, we shouldn't over-complicate the naming regular expression by trying to cover every edge case.

**Comment by BraisGabin** [comment]
I don't see this as an edge case. This is even pointed out by the coding convention that we know it has a lot of grey areas where they don't define all the cases.

I agree that adding a new configuration is probably not the best way. And I know that we have bigger issues in the naming implementation: #3150

I think that we could implement the idea proposed by @yogurtearl with regex:
`[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*[A-Z]{0,3}` (I didn't check it so be careful if you use this regex)

**Comment by schalkms** [comment]
> This is even pointed out by the coding convention and we know it has a lot of grey areas where they don't define all the cases.

That’s exactly the problem I see. This is for sure on of the controversial points in the guideline.

> I think that we could implement the idea proposed by @yogurtearl with regex:
[a-z][a-z0-9]*([A-Z]{1,3}[a-z0-9]+)*[A-Z]{0,3} (I didn't check it so be careful if you use this regex)

@yogurtearl can you please try that out for your use case? 

Furthermore, the change for the default config would be kind of breaking, since exisiting code would get flagged with the new default config in the next detekt version. Anyhow, I still suggest to leave the choice up to the user.
